src/CpmsClient/Client/ClientOptions.php

```diff
@@ -1,200 +1,131 @@
 <?php
+
 namespace CpmsClient\Client;
 
+use Exception;
 use Laminas\Stdlib\AbstractOptions;
 
 /**
  * Class ClientOptions
+ * @extends AbstractOptions<iterable>
  *
  * @package CpmsClient\Client
  */
 class ClientOptions extends AbstractOptions
 {
-    /** @var int */
-    protected $version = 1;
-    /** @var  string */
-    protected $clientId;
-    /** @var  string */
-    protected $clientSecret;
-    /** @var  string */
-    protected $userId;
-    /** @var array */
-    protected $endPoints = array();
-    /** @var  string */
-    protected $customerReference;
-    /** @var  string */
-    protected $grantType;
-    /** @var int */
-    protected $timeout = 30;
-
+    protected int $version = 1;
+    protected ?string $clientId = null;
+    protected ?string $clientSecret = null;
+    protected ?string $userId = null;
+    protected array $endPoints = array();
```

Comment:
Minor: We can swap array() for []
```suggestion
    protected array $endPoints = [];
```

Reply:
done